vault/resource_aws_secret_backend_role.go

```diff
@@ -36,10 +36,9 @@ func awsSecretBackendRoleResource() *schema.Resource {
 				Description: "The path of the AWS Secret Backend the role belongs to.",
 			},
 			"policy_arns": {
-				Type:          schema.TypeSet,
-				Optional:      true,
-				ConflictsWith: []string{"policy", "policy_arn", "role_arns"},
```

Comment:
Hi! I'd be fine with pulling `role_arns` out here. However, I'm not so sure we should pull out the rest. Adding these deprecated parameters to `ConflictsWith` here is a normal part of the best practices around deprecation located [here](https://www.terraform.io/docs/extend/best-practices/deprecations.html).

Reply:
Hi!

That does seem reasonable as we're still not at the next major version bump.